### ISSUE TITLE
Chat Docks, Loot Roll Window, Chat resize.

### ISSUE THREAD

**Issue by crowiee92** [joev/SVUI-Temp#31]
Hi, let me start off by saying thank you for temporarily taking over this addon and giving it fixes until the creator comes back. <3 

I've noticed a few issues I think a lot of people would like to have changed:
**-The Bottom Left Dock**, whenever I set up my chat (I like to have multiple tabs, like whisper, guild, trade, general/local all in different tabs) and reload or relog or anything that has the same effect will reset my chat tabs, it doesn't delete them luckily but it's still annoying to have to enable them every single time I do a relog/reload.
**-Loot Roll Window** causes an LUI error (didn't copy the error sorry)
**-Chat** is too big and I can't (or at least I can't find an option to) resize it, I'd like it better personally if the chat window was a completely seperate window and not part of the bottom left dock.. then just put an arrow at the bottom right which you click and drag to resize the window to your heart's content (like default chat ui and chatter, unlock window and click the corner and drag to resize) 'cause right now it's just getting in the way of everything, it takes up way too much space on the screen so I'm forced to use chatter for the time being.
**-** **Quest Tracker Window** starts flashing whenever I track a quest, also it'd be nice to have an arrow in the quest tracker itself that points to the quest you're currently tracking, and quest items don't appear in the window.
**-** **Reputation Bar** still gives error.
**-** **Also**, and this might be the most important one: can we have an option to have an **Artifact Power Bar**? since I'm level cap and therefor don't need the xp bar anymore I'd like to replace that spot with an artifact bar (same design as xp bar would be cool) I'm sorry if this option is already in there and I just overlooked it like a derp but I can't find it, it seems such an obvious thing too since AP is arguably the most important thing in this expansion.

I hope you can fix these issues.

**Comment by klepp0906** [comment]
- Quest Tracker Window starts flashing whenever I track a quest, also it'd be nice to have an arrow in the quest tracker itself that points to the quest you're currently tracking, and quest items don't appear in the window.
- Reputation Bar still gives error.
- Also, and this might be the most important one: can we have an option to have an Artifact Power Bar? since I'm level cap and therefor don't need the xp bar anymore I'd like to replace that spot with an artifact bar (same design as xp bar would be cool) I'm sorry if this option is already in there and I just overlooked it like a derp but I can't find it, it seems such an obvious thing too since AP is arguably the most important thing in this expansion.

these!!  the artifact is a great idea!  the current/needed power for the next trait would be helpful!

the chat i have roughly the same issues and made a post about them a few days ago.   he's on it.  updates pushed daily so keep checking the master.

**Comment by drveoj** [comment]
I got sick of just fixing bugs and implemented the Artifact Bar ;) 

Hope it works for you! :)

**Comment by klepp0906** [comment]
Eager to  check it out in the morning!  Did you just implement a bar or a bar & a numerical representation (aka the equivalent of what rep and experience options allow either or)

Only ask cause I prefer the numerical current/required type display over a bar.

Also I can't say it enough times.  Thank you for all the work your doing.  I know it's a hot mess but your slowly bringing it back to life and it is the king of uis in my book :p

Btw, instead of re opening another issue - with the one you closed regarding the protected function calling - you think it's related to another add on?  So what happens if It is?  Do I take it to that add on developer for a fix or at that point is it fixable on the svui end?

My taint log didn't help eh?

Also - based on the lua error, can you point me toward what type of add on it could be or?

Tyty!

**Comment by klepp0906** [comment]
artifact bar working well btw :)

**Comment by Murphy107j** [comment]
I would like to add a bug, when I make a new chat window for party chat it disappears when i reload ui or go through a load screen

**Comment by drveoj** [comment]
@Murphy107j Certainly sounds like a bug. if you could raise that as a separate issue that would be great.

**Comment by drveoj** [comment]
> -The Bottom Left Dock, whenever I set up my chat (I like to have multiple tabs, like whisper, guild, trade, general/local all in different tabs) and reload or relog or anything that has the same effect will reset my chat tabs, it doesn't delete them luckily but it's still annoying to have to enable them every single time I do a relog/reload.

Hi @crowiee92 Can you describe to me what you mean by "reset"? I suspect you, @klepp0906 and @Murphy107j are all describing the same issue in slightly different ways, none of which I quite understand as my chat windows appear to me to be operating properly (but then I'm most likely not doing what you're doing that makes you say they're not working)

**Comment by crowiee92** [comment]
What I mean by reset is anything that reloads the UI, like exiting and restarting the game, or logging out of a character and logging back in, or simply reloading the UI using the /reload method. 
I like to use multiple tabs for my chat, having all channels in different tabs, it doesn't delete them after a reload, but it resets them to their original position on each reload (which for most of them is simply turning them off since they're not all enabled by default)

**Comment by dasmall80** [comment]
Adding to this as I believe it is all related.  I also get the chat tabs disappearing on ui reload, but I have noticed that the tabs seem to exist but behind the other tabs.  I will see overlapping text that can't be cleared until the missing tabs are recreated.  Similar thing happens if I try to remove fight-o-matic, the tab is gone, but all of its functions are sitting behind everything else after a ui reload. 
![fightomatic](https://cloud.githubusercontent.com/assets/22858732/19412173/f45f7f60-92dd-11e6-8d9c-6a066fc57b2a.PNG)

**Comment by crowiee92** [comment]
You're right, after I read your comment and went online to check for myself and it is indeed as you said.
I noticed overlapping text tabs and when I enabled the fight-o-matic addon (I never use it so I never have the addon in the first place) but disable the dock then the dock disappears but all it's functions stay and overlap the other docks.

Thanks for pointing this out, now that we know exactly what happens it may be easier to fix.
